syntax = "proto2";
package wal;

message WalRecord {
  optional int64 term = 1;
  optional bytes log = 2;
}

message WalStreamMeta {
  repeated bytes stream_uuids = 1;
}

message WalStreamNextLogIdMeta {
  optional uint64 next_log_id = 1;
}

message WalStreamLowerBoundMeta {
  optional uint64 lower_bound = 1;
}

message WalStreamUpperBoundMeta {
  optional uint64 upper_bound = 2;
}
